app/scripts/controllers/visualizerModal.js

```diff
@@ -0,0 +1,204 @@
+      $scope.settings.visualizer.mediators.push({ mediator: $scope.visualizer.addSelectMediator.urn, name: $scope.visualizer.addSelectMediator.name, display: $scope.visualizer.addSelectMediator.name });
+      $scope.visualizer.addSelectMediator = null;
+    };
+
+    $scope.addComponent = function(){
+      if( $scope.visualizer.addComponent.eventType ){
+        $scope.settings.visualizer.components.push({ eventType: $scope.visualizer.addComponent.eventType, eventName: $scope.visualizer.addComponent.eventName, display: $scope.visualizer.addComponent.display });
+        $scope.visualizer.addComponent.eventType = '';
+        $scope.visualizer.addComponent.eventName = '';
+        $scope.visualizer.addComponent.display = '';
+      }
+    };
+
+    $scope.removeComponent = function(index){
+      $scope.settings.visualizer.components.splice(index, 1);
+    };
+
+    $scope.removeChannel = function(index){
+      $scope.settings.visualizer.channels.splice(index, 1);
+    };
+
+    $scope.removeMediator = function(index){
+      $scope.settings.visualizer.mediators.splice(index, 1);
+    };
+
+    $scope.validateFormSettings = function(settings, callback){
+      // reset hasErrors alert object
+      Alerting.AlertReset('hasErrors');
+
+      // clear timeout if it has been set
+      $timeout.cancel( $scope.clearValidation );
+
+      $scope.ngError.hasErrors = false;
+
+      // required fields validation
+      if(!settings.name){
+        $scope.ngError.hasNoName = true;
+        $scope.ngError.hasErrors = true;
+      } else {
+
+        // the visualizer name must be unique
+        var result = visualizers.filter(function (obj){
+          return obj.name === settings.name;
+        });
+
+        if(result.length > 0){
+          $scope.ngError.nameNotUnique = true;
+          $scope.ngError.hasErrors = true; 
+        }
+      }      
+
+      if(settings.components === undefined || settings.components.length === 0){
+        $scope.ngError.hasErrors = true;
+        $scope.ngError.hasNoComponents = true;
+      }
+
+      if(settings.mediators === undefined || settings.mediators.length === 0){
+        $scope.ngError.hasErrors = true;
+        $scope.ngError.hasNoMediators = true;
+      }
+
+      if(settings.channels === undefined || settings.channels.length === 0){
+        $scope.ngError.hasErrors = true;
+        $scope.ngError.hasNoChannels = true;
+      }
+
+      if ( $scope.ngError.hasErrors ){
+        $scope.clearValidation = $timeout(function(){
+          // clear errors after 5 seconds
+          $scope.ngError = {};
+          Alerting.AlertReset('hasErrors');
+        }, 5000);
+        Alerting.AlertAddMsg('hasErrors', 'danger', 'There are errors on the form.');
+        callback('Error: Form has errors');
+      } else {
+        callback();
+      }
+    };
+
+    var notifyUser = function(){
+      // reset backing object and refresh users list
+      Notify.notify('channelsChanged');
```

Comment:
This is probably from existing code, but `channelChanged` seems to wrong. Is this even used in the visualizer list controller?